### PR TITLE
feat: CO-56 New Q Mastercard theme

### PR DESCRIPTION
**Pull request by ghost**
## PR Area

- [x] Frontend
- [ ] Testing
- [ ] Other: 

## Type of change

_What kind of change does this PR introduce?_

- [ ] Bugfix
- [x] Feature
- [ ] Other: 

## Description
* Adds ned `darkMode` paramater
* New options for buttons: rounded, bold, `md` size
* Changes dialog slot from `header` -> `title` and add new `header` slot above the body
* QMC light and dark theme
* Fixes IconClose opacity properly being passed down only from the parent
* Existing mastercard content module
* Allow configuring `WidgetContent` with parameters

## Security & risks

- [ ] Adds new third party libraries
- [ ] Adds new `.env` variables
- [ ] Integrates with Third-party apis (dependency must be resilient)
- [ ] Has specific [OWASP security concerns](https://owasp.org/www-project-top-ten/) - list them: (also see [Fixing OWASP Top 10](https://nodegoat.herokuapp.com/tutorial))

## Pre-publish checklist

- [ ] I have assigned myself to the ticket
- [x] I have performed a self-review of my code
- [ ] I have performed relevant browser testing
- [ ] I have performed keyboard testing of new or modified interactive components

#### Evidence screenshots
Screen Shot 2021-09-22 at 6 46 18 pm
Screen Shot 2021-09-22 at 6 46 25 pm

## Reviewer checklist

- [ ] I have run the code (if applicable)
- [ ] I have reviewed and run any tests (if applicable)

### CODE REVIEW

**Review by jpyepez-roam — APPROVED** [review]
LGTM! I really like the multi-slot setup there.

File: src/modules/DialogOverlay.vue [hunk 1]
```diff
@@ -5,8 +5,11 @@
     :button-close-label="buttonCloseLabel"
     @toggle-dialog="toggleDialog"
   >
+    <template #title>
+      <slot name="title">Interest-free Payment</slot>
+    </template>
     <template #header>
-      <slot name="header" />
+      <MastercardExisting />
```
Comment (arjun-at-roam) [inline src/modules/DialogOverlay.vue]: would it be better to keep the slot and pass in MastercardExisting from parent? It's not a blocker but will probably have to be tackled as part of Humm90 theme work.
image

File: src/modules/MastercardExisting.vue [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    margin-bottom: 2px;
+  }
+
+  &__list {
+    margin: 0;
+    padding-left: 5px;
+    list-style-position: inside;
+  }
+}
+</style>
```
Comment (arjun-at-roam) [inline src/modules/MastercardExisting.vue:52]: LEGEND! thanks for doing the overlay as well! 🎉

**Review by arjun-at-roam — APPROVED** [review]

File: src/modules/DialogOverlay.vue (hunk 1 above)
Comment (ghost) [reply src/modules/DialogOverlay.vue]: Yep, good point! Feel free to do that if you want to! This is your baby now 😁

**Pull request merged**